test/test_ConfigFileParser.cpp

```diff
@@ -0,0 +1,270 @@
+            } catch (const std::exception& e) {
+                EXPECT_STREQ("Semicolon missing", e.what());
+                throw;
+            }
+        },
+        std::runtime_error);
+}
+
+TEST_F(InvalidConfigFileTests, ListenDirectiveContainsTooHighIpAddress)
+{
+    EXPECT_THROW(
+        {
+            try {
+                m_configFileParser.parseConfigFile("config_files/listen_invalid_ip_high.conf");
+            } catch (const std::exception& e) {
+                EXPECT_STREQ("Invalid ip address", e.what());
+                throw;
+            }
+        },
+        std::runtime_error);
+}
+
+TEST_F(InvalidConfigFileTests, ListenDirectiveContainsIpAddressWithMissingDot)
+{
+    EXPECT_THROW(
+        {
+            try {
+                m_configFileParser.parseConfigFile("config_files/listen_invalid_ip_missing_dot.conf");
+            } catch (const std::exception& e) {
+                EXPECT_STREQ("Invalid ip address", e.what());
+                throw;
+            }
+        },
+        std::runtime_error);
+}
+
+TEST_F(InvalidConfigFileTests, ListenDirectiveContainsTooHighPort)
+{
+    EXPECT_THROW(
+        {
+            try {
+                m_configFileParser.parseConfigFile("config_files/listen_invalid_port_high.conf");
+            } catch (const std::exception& e) {
+                EXPECT_STREQ("Invalid port", e.what());
+                throw;
+            }
+        },
+        std::runtime_error);
+}
+
+TEST_F(InvalidConfigFileTests, ListenDirectiveContainsTooLowPort)
+{
+    EXPECT_THROW(
+        {
+            try {
+                m_configFileParser.parseConfigFile("config_files/listen_invalid_port_low.conf");
+            } catch (const std::exception& e) {
+                EXPECT_STREQ("Invalid port", e.what());
+                throw;
+            }
+        },
+        std::runtime_error);
+}
+
+TEST_F(InvalidConfigFileTests, RootDirectiveContainsNoPath)
+{
+    EXPECT_THROW(
+        {
+            try {
+                m_configFileParser.parseConfigFile("config_files/root_no_path.conf");
+            } catch (const std::exception& e) {
+                EXPECT_STREQ("'root' directive has no value", e.what());
+                throw;
+            }
+        },
+        std::runtime_error);
+}
+
+TEST_F(InvalidConfigFileTests, RootDirectiveContainsMultipleRootPaths)
+{
+    EXPECT_THROW(
+        {
+            try {
+                m_configFileParser.parseConfigFile("config_files/root_multiple_paths.conf");
+            } catch (const std::exception& e) {
+                EXPECT_STREQ("More than one root path", e.what());
+                throw;
+            }
+        },
+        std::runtime_error);
+}
+
+/**
+ * @brief Tests for a valid file
+ *
+ * Checks if the following configuration are seen as valid:
+ * 1. A standard valid file
+ * 2. Listen directive contains only ip
+ * 3. Listen directive contains only port
+ * 4. Listen contains ip and port
+ */
+
+TEST_F(ValidConfigFileTests, ValidFile)
```

Comment:
I think these should also check the contents of the parsed configfile. Right now we only check if the Parser does not throw but we never validate that the parser parses correctly ;)

Reply:
True. Added to the validFile test checks for the ip address and the port in the map listen and for the root in the string root (for the server block).